src/util/selectionUtils.ts

```diff
@@ -42,3 +42,22 @@ export function selectionFromRange(isReversed: boolean, range: Range) {
   const { start, end } = range;
   return isReversed ? new Selection(end, start) : new Selection(start, end);
 }
+
+/**
+ * Shrink given range to fit text content.
+ * If the given range has no text content the given range will be returned unchanged.
```

Comment:
What does "no text content" mean?  Does that mean "if the range contains only whitespace"?

Reply:
That or empty range